### PR TITLE
Refactor AdorsysIntegServiceProvider

### PR DESCRIPTION
**Pull request by berezkin88**

### CODE REVIEW

File: adapters/adorsys-adapter/src/main/java/de/adorsys/xs2a/adapter/adorsys/AdorsysSigningHeadersInterceptor.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+
+public class AdorsysSigningHeadersInterceptor implements Interceptor {
+
+    private final RequestSigningInterceptor signingInterceptor;
+    private final boolean requestSigningEnabled =
+        Boolean.parseBoolean(AdapterConfig.readProperty("adorsys.request_signing.enabled", "false"));
+
+    public AdorsysSigningHeadersInterceptor(RequestSigningInterceptor signingInterceptor) {
```
Comment (spetrychenko) [inline adapters/adorsys-adapter/src/main/java/de/adorsys/xs2a/adapter/adorsys/AdorsysSigningHeadersInterceptor.java:18]: I propose to the Keystore as a parameter and hide the creation of RequestSigningInterceptor

File: adapters/adorsys-adapter/src/main/java/de/adorsys/xs2a/adapter/adorsys/AdorsysIntegServiceProvider.java [hunk 2]
```diff
@@ -17,63 +17,45 @@
...
     @Override
     public PaymentInitiationService getPaymentInitiationService(Aspsp aspsp,
                                                                 HttpClientFactory httpClientFactory,
                                                                 Pkcs12KeyStore keyStore,
                                                                 LinksRewriter linksRewriter) {
+        adorsysSigningHeadersInterceptor = new AdorsysSigningHeadersInterceptor(new RequestSigningInterceptor(keyStore));
         return new BasePaymentInitiationService(aspsp,
                                                 httpClientFactory.getHttpClient(getAdapterId()),
-                                                keyStore != null ? getInterceptor(keyStore) : null,
+                                                Arrays.asList(oauthHeaderInterceptor, adorsysSigningHeadersInterceptor),
```
Comment (spetrychenko) [inline adapters/adorsys-adapter/src/main/java/de/adorsys/xs2a/adapter/adorsys/AdorsysIntegServiceProvider.java:45]: and what about the case when a keystore is null?

Comment (berezkin88) [reply adapters/adorsys-adapter/src/main/java/de/adorsys/xs2a/adapter/adorsys/AdorsysIntegServiceProvider.java:45]: As I understood, this null checking was here only to make WireMock tests running. I tweaked the test a bit to negate this checking.
Can it ever be null?
It's not possible to start the Adapter app without a KeyStore. Besides, this is the only place in Adapter where KeyStore is checked for null.

File: adapters/adorsys-adapter/src/main/java/de/adorsys/xs2a/adapter/adorsys/AdorsysSigningHeadersInterceptor.java (hunk 1 above)
Comment (berezkin88) [reply adapters/adorsys-adapter/src/main/java/de/adorsys/xs2a/adapter/adorsys/AdorsysSigningHeadersInterceptor.java:18]: The tests should be adjusted then. Working on it.

File: adapters/adorsys-adapter/src/main/java/de/adorsys/xs2a/adapter/adorsys/AdorsysSigningHeadersInterceptor.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        return requestBuilder;
+    }
+
+    private void setDateHeader(Request.Builder requestBuilder) {
+        requestBuilder.header(RequestHeaders.DATE,
+            DateTimeFormatter.RFC_1123_DATE_TIME.format(ZonedDateTime.now()));
+    }
+
+    private void setSignatureCertificateHeader(Request.Builder requestBuilder) {
+        signingInterceptor.preHandle(requestBuilder);
```
[the remark this answers is not in the document]
Comment (berezkin88) [reply adapters/adorsys-adapter/src/main/java/de/adorsys/xs2a/adapter/adorsys/AdorsysSigningHeadersInterceptor.java:38]: ok

File: adapters/adorsys-adapter/src/main/java/de/adorsys/xs2a/adapter/adorsys/AdorsysSigningHeadersInterceptor.java (hunk 1 above)
Comment (berezkin88) [reply adapters/adorsys-adapter/src/main/java/de/adorsys/xs2a/adapter/adorsys/AdorsysSigningHeadersInterceptor.java]: done

File: adapters/adorsys-adapter/src/main/java/de/adorsys/xs2a/adapter/adorsys/AdorsysIntegServiceProvider.java (hunk 2 above)
Comment (berezkin88) [reply adapters/adorsys-adapter/src/main/java/de/adorsys/xs2a/adapter/adorsys/AdorsysIntegServiceProvider.java]: done

File: adapters/adorsys-adapter/src/main/java/de/adorsys/xs2a/adapter/adorsys/AdorsysIntegServiceProvider.java [hunk 4]
```diff
@@ -45,35 +41,25 @@ public PaymentInitiationService getPaymentInitiationService(Aspsp aspsp,
...
     public AccountInformationService getAccountInformationService(Aspsp aspsp,
                                                                   HttpClientFactory httpClientFactory,
                                                                   Pkcs12KeyStore keyStore,
                                                                   LinksRewriter linksRewriter) {
-        return new AdorsysAccountInformationService(aspsp, httpClientFactory.getHttpClient(getAdapterId()),
-                                                    getInterceptor(keyStore), linksRewriter);
+        return new AdorsysAccountInformationService(aspsp,
+                                                    httpClientFactory.getHttpClient(getAdapterId()),
+                                                    Arrays.asList(oauthHeaderInterceptor,
+                                                        getAdorsysSigningHeadersInterceptor(keyStore)),
```
Comment (spetrychenko) [inline adapters/adorsys-adapter/src/main/java/de/adorsys/xs2a/adapter/adorsys/AdorsysIntegServiceProvider.java:61]: I would suggest do not put null into the list

Comment (berezkin88) [reply adapters/adorsys-adapter/src/main/java/de/adorsys/xs2a/adapter/adorsys/AdorsysIntegServiceProvider.java]: refined

**Review by spetrychenko — APPROVED** [review]

**Pull request merged**